File: java/Iterator/src/iteratorpattern/CountryDetails.java
package iteratorpattern;

import java.util.Iterator;

public class CountryDetails {
	public static void main(String[] args){
		Iterator it;
		System.out.println("Country Set");
		it = new CountrySet().__________; //>> Blank 2
		displayCountries(it);
		System.out.println("Country Array");
		it = new CountryArray().___________; //>> Blank 3
		displayCountries(it);
		System.out.println("Country List");
		it = new CountryList().____________; //>> Blank 4
		displayCountries(it);		
	}
	public static void displayCountries(Iterator it){
		while (it.hasNext()){
			Country c= ____________; //>> Blank 5
			System.out.println("Country Name = "+c.getName()+" Country Area = "+c.getArea()+" Country Population = "+c.getPopulation() );
		}
	}
}

File: ruby/Iterator/.idea/country_details.rb
require 'set'
class Country
  def initialize name, area, population
    @name = name;
    @area = area;
    @population = population;
  end
  def getName
    @name;
  end
  def getArea
    @area;
  end
  def getPopulation
    @population;
  end
end

class CountryArray
  def initialize
    @countryArray = [Country.new("Australia",297,23),Country.new("Germany",14,81), Country.new("France",25,66)]
  end
  def getCountryArray
    @countryArray
  end
end

class CountrySet
  def initialize
    @countrySet = Set[Country.new("India",120,1252),Country.new("United States",380,316), Country.new("China",370,1357)]
  end
  def getCountrySet
    @countrySet
  end
end

class CountryDetails
  def self.displayCounries(group)
    group.each do|country|
      puts "Country Name = #{country.getName} Country Area = #{country.getArea} Country Population = #{country.getPopulation}"
    end
  end
end

countryArray = CountryArray.new.getCountryArray
countrySet = CountrySet.new.getCountrySet

CountryDetails.displayCounries(countryArray)
CountryDetails.displayCounries(countrySet)
